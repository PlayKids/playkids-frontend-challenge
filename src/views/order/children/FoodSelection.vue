<template>
  <div>
    <c-items-picker
      :list="foodList"
      @onItemChosen="item => handleChosenItem(item)"
    />
    <div class="actions-container">
      <c-button @click="back">Back</c-button>
      <c-button @click="next">Next</c-button>
    </div>
  </div>
</template>

<script>
import { mapGetters, mapState } from 'vuex';
import CItemsPicker from '@/components/organisms/c-items-picker/CItemsPicker';
import CButton from '@/components/atoms/c-button/CButton';

export default {
  name: 'ItemSelection',

  components: {
    CItemsPicker,
    CButton
  },

  computed: {
    ...mapGetters('restaurantMenu', ['foodList']),
    ...mapState('order', ['newOrder'])
  },

  created() {
    if (!this.newOrder.customerIdentification) {
      this.$router.push('/order/customer');
      return;
    }

    this.$store.commit('order/setCurrentStep', 'Choose your foods');
    if (!this.foodList.length) this.$store.dispatch('restaurantMenu/get');
  },

  methods: {
    handleChosenItem(item) {
      this.$store.commit('order/setItem', item);
      this.$store.commit('restaurantMenu/setItemSelected', {
        itemId: item.id,
        selected: true
      });
    },

    next() {
      this.$router.push('/order/drink');
    },
    back() {
      this.$router.push('/order/customer');
    }
  }
};
</script>

<style scoped lang="scss">
@use '../../../assets/styles/sizes' as *;

.actions-container {
  display: flex;
  justify-content: space-between;
  margin-top: $size-md;
}
</style>
