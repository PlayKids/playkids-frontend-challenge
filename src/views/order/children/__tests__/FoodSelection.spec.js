import { mount, createLocalVue } from '@vue/test-utils';
import FoodSelection from '../FoodSelection.vue';
import Vuex from 'vuex';

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.filter('currency', value => value);

describe('FoodSelection view', () => {
  let store;

  beforeEach(() => {
    store = new Vuex.Store({
      mutations: {
        setAppLoading: () => {}
      },
      modules: {
        order: {
          namespaced: true,
          state: {
            newOrder: {
              customerIdentification: 'John',
              items: []
            }
          },
          mutations: {
            setCurrentStep: () => {}
          }
        },
        restaurantMenu: {
          namespaced: true,
          getters: {
            foodList: () => [
              {
                name: 'Fish',
                price: 8.9
              }
            ]
          },
          actions: {
            get: () => []
          }
        }
      }
    });
  });

  it('renders correctly', () => {
    const wrapper = mount(FoodSelection, { localVue, store });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders fish product', async () => {
    const wrapper = mount(FoodSelection, { localVue, store });
    const itemCard = wrapper.find('.cards-container .card').text();

    expect(itemCard).toContain('Fish');
  });
});
