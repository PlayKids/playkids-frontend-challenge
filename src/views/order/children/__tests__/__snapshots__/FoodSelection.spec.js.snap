// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoodSelection view renders correctly 1`] = `
<div>
  <div class="input"><label class="input__label">Search your food</label> <input class="input__field">
    <!---->
  </div>
  <div>
    <div class="cards-container">
      <div class="card clickable">
        <p>Fish</p>
        <p></p>
        <!---->
        <p>8.9</p>
      </div>
    </div>
    <transition-stub name="modal">
      <!---->
    </transition-stub>
  </div>
  <div class="actions-container"><button class="button">Back</button> <button class="button">Next</button></div>
</div>
`;
